### PR TITLE
Feature/claim airdrop

### PR DESCRIPTION
**Pull request by wdragen**

### CODE REVIEW

File: contracts/AirdropClaim.sol [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+pragma solidity 0.7.6;
+
+import "./libraries/Ownable.sol";
+import "./interfaces/IBEP20.sol";
+
+
+contract AirdropClaim is Ownable {
+
+    uint public claimExpiredAt;
```
Comment (kingter) [inline contracts/AirdropClaim.sol:11]: Would it better to use `private` for better security?

save comments apply attributes below.

File: contracts/AirdropClaim.sol [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    constructor(address airdropToken_) {
+        require(airdropToken_ != address(0));
+        airdropTokenAddress = airdropToken_;
+    }
+
+    receive () external payable {}
+    fallback () external payable {}
+
+    function pullFunds(address tokenAddress_) onlyOwner public {
```
Comment (kingter) [inline contracts/AirdropClaim.sol:34]: How about adding a few words as comments to explain the purpose of the function?

File: contracts/AirdropClaim.sol [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        delete amount;
+
+        emit Claimed(claimId++, msg.sender, amount);
+
+        IBEP20 token = IBEP20(airdropTokenAddress);
+        require(token.transfer(msg.sender, amount), "Token transfer failed");
+
+        return true;
+    }
+}
```
Comment (kingter) [inline contracts/AirdropClaim.sol]: Might be better to add one more empty line at the end of the file?

**Review by kingter — COMMENTED** [review]
Looks good to me!
Thanks so much @songfrank quick turn around and implement the airdrop function!
Just a few minor comments, please kindly take a look to see whether they make sense or not.

File: contracts/AirdropClaim.sol [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        require(airdropToken_ != address(0));
+        airdropTokenAddress = airdropToken_;
+    }
+
+    receive () external payable {}
+    fallback () external payable {}
+
+    function pullFunds(address tokenAddress_) onlyOwner public {
+        if (tokenAddress_ == address(0)) {
+            msg.sender.transfer(address(this).balance);
```
Comment (kingter) [inline contracts/AirdropClaim.sol:36]: Could you help to explain a bit more the purpose of this logic? 
How about check the tokenAddress_ should not be null? `require(tokenAddress_ != `address(0));`

File: contracts/AirdropClaim.sol [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        require(airdropToken_ != address(0));
+        airdropTokenAddress = airdropToken_;
+    }
+
+    function setClaimExpiredAt(uint expireTimestamp_) onlyOwner public {
+        require(expireTimestamp_ > block.timestamp, "The expireTimestamp has passed");
+        claimExpiredAt = expireTimestamp_;
+    }
+
+    function setupWhitelist(address[] calldata candidates_, uint[] calldata values_) onlyOwner public returns (bool){
```
Comment (LeoHChen) [inline contracts/AirdropClaim.sol:53]: So the Owner can overwrite the candidate list if this function is called 2nd times?

File: contracts/AirdropClaim.sol (hunk 2 above)
Comment (wdragen) [reply contracts/AirdropClaim.sol:34]: done

File: contracts/AirdropClaim.sol (hunk 3 above)
Comment (wdragen) [reply contracts/AirdropClaim.sol]: done

File: contracts/AirdropClaim.sol (hunk 5 above)
Comment (wdragen) [reply contracts/AirdropClaim.sol:53]: 是的，owner可以覆盖修改之前设置过的名单。每个地址的可领取额度需要在链下计算好再调用该函数。

File: contracts/AirdropClaim.sol (hunk 1 above)
Comment (wdragen) [reply contracts/AirdropClaim.sol:11]: 我觉得public 方便直接读取该值来检查是否正确。

**Review by kingter — APPROVED** [review]

File: contracts/AirdropClaim.sol (hunk 5 above)
Comment (kingter) [reply contracts/AirdropClaim.sol:53]: Let's resolved the comments and merge the code first. we could file another PR later if there is any change that would be considered needed.

**Pull request merged**